src/NuGet.Clients/NuGet.VisualStudio.Internal.Contracts/Formatters/FloatRangeFormatter.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using MessagePack;
+using MessagePack.Formatters;
+using Microsoft;
+using NuGet.Versioning;
+
+namespace NuGet.VisualStudio.Internal.Contracts
+{
+    internal sealed class FloatRangeFormatter : IMessagePackFormatter<FloatRange?>
+    {
+        private const string FloatBehaviorPropertyName = "floatbehavior";
+        private const string MinVersionPropertyName = "minversion";
+        private const string ReleasePrefixPropertyName = "releaseprefix";
+
+        internal static readonly IMessagePackFormatter<FloatRange?> Instance = new FloatRangeFormatter();
+
+        private FloatRangeFormatter()
+        {
+        }
+
+        public FloatRange? Deserialize(ref MessagePackReader reader, MessagePackSerializerOptions options)
+        {
+            if (reader.TryReadNil())
+            {
+                return null;
+            }
+
+            // stack overflow mitigation - see https://github.com/neuecc/MessagePack-CSharp/security/advisories/GHSA-7q36-4xx7-xcxf
+            options.Security.DepthStep(ref reader);
+
+            try
+            {
+                NuGetVersionFloatBehavior? floatBehavior = null;
```

Comment:
Using the original string to construct a full blown version range might've been an easier serialization.